enum.c

```diff
@@ -132,21 +132,23 @@ enum_grep0(VALUE obj, VALUE pat, VALUE test)
 }
 
 /*
- *  call-seq:
- *     enum.grep(pattern)                  -> array
- *     enum.grep(pattern) { |obj| block }  -> array
- *
- *  Returns an array of every element in <i>enum</i> for which
- *  <code>Pattern === element</code>. If the optional <em>block</em> is
- *  supplied, each matching element is passed to it, and the block's
- *  result is stored in the output array.
- *
- *     (1..100).grep 38..44   #=> [38, 39, 40, 41, 42, 43, 44]
- *     c = IO.constants
- *     c.grep(/SEEK/)         #=> [:SEEK_SET, :SEEK_CUR, :SEEK_END]
- *     res = c.grep(/SEEK/) { |v| IO.const_get(v) }
- *     res                    #=> [0, 1, 2]
- *
+ * call-seq:
+ *   grep(object) -> new_array
+ *   grep(object) {|element| ... } -> new_array
+ *
+ * Returns a new \Array selected by a given object or block.
```

Comment:
why "new"? (the receiver needs not be an array)

Reply:
Isn't the return always a new Array?

Reply:
It is a new Array 👍

Reply:
@BurdetteLamar @marcandre

compare to

```
array.join -> string
array.reverse -> new_array
```
In both cases the returned object is newly created, but "new_thing" is usually used when the receiver is a "thing" and the method returns a different "thing".